tests/syscall/mount/mount-immutables.bats

```diff
@@ -0,0 +1,596 @@
+#!/usr/bin/env bats
+
+#
+# Verify trapping & emulation on "mount" and "unmount2" syscalls
+#
+
+load ../../helpers/run
+load ../../helpers/syscall
+load ../../helpers/docker
+load ../../helpers/environment
+load ../../helpers/mounts
+load ../../helpers/sysbox-health
+
+skipTest=0
+
+function teardown() {
+  sysbox_log_check
+}
+
+# Test to verify immutable mounts within a sys container.
+#
+# Note: a sys container immutable mount is a mount that is setup at container
+# creation time.
+#
+
+# Testcase #1.
+#
+# Ensure immutable mounts can't be unmounted from inside the container
+@test "immutable mount can't be unmounted" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  local immutable_mounts=$(list_container_mounts ${syscont} "0" "/")
+  run empty_list ${immutable_mounts}
+  [ "$status" -ne 0 ]
+
+  for m in $immutable_mounts; do
+    # Skip /proc and /sys since these are special mounts (we have dedicated
+    # tests that cover unmounting ops).
+    if [[ ${m} =~ "/proc" ]] || [[ ${m} =~ "/proc/*" ]] ||
+        [[ ${m} =~ "/sys" ]] || [[ ${m} =~ "/sys/*" ]]; then
+        continue
+    fi
+
+    printf "\ntesting unmount of immutable mount ${m}\n"
+
+    docker exec ${syscont} sh -c "umount ${m}"
+    [ "$status" -ne 0 ]
+  done
+
+  local immutable_mounts_after=$(list_container_mounts ${syscont} "0" "/")
+  [[ $immutable_ro_mounts == $immutable_ro_mounts_after ]]
+
+  docker_stop ${syscont}
+}
+
+# Testcase #2.
+#
+# Ensure that a read-only immutable mount can't be remounted as read-write
+# inside the container.
+@test "immutable ro mount can't be remounted rw" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  local immutable_ro_mounts=$(list_container_ro_mounts ${syscont} "0" "/")
+  run empty_list ${immutable_ro_mounts}
+  [ "$status" -ne 0 ]  
+
+  for m in $immutable_ro_mounts; do
+    printf "\ntesting rw remount of immutable ro mount ${m}\n"
+
+    docker exec ${syscont} sh -c "mount -o remount,bind,rw ${m}"
+    [ "$status" -ne 0 ]
```

Comment:
I would add a `docker exec ${syscont} sh -c "touch ${m}"` to verify the mount continues to indeed be read-only (i.e., the touch will fail).